azerothcore/azerothcore-wotlk
# fix(Scripts/Scarlet monastery) Whitemane and Mograine's battle scripts are more similar to Blizzard

## Changes Proposed:
- The church door needs a key to open
-  When Whitemane's  HP is below 50%, hypnotize the player and run to the vicinity of Mograine's corpse, waiting for 3 seconds left in the player's hypnotic time to release the Resurrection Spell, ~~and use a saluting emoji to mograine after the release is completed~~
-  Mograine uses SPELL_LAY_ON_HAND spells to heal Whitemane after resurrection, ~~and uses the "Roar" emoji after healing~~
-  Whitemane can't kill her until she has unleashed her resurrection
- If Whitemane's dies and Mograine leaves the battle alive, Mograine does not reset
## Issues Addressed:

- Closes 

## SOURCE:

~~Blizzard servers~~
https://youtu.be/lFwkkXyE7Hg  
This is a WLK video I can find from 13 years ago
3:56 sec The lower right foot shows that the key needs to be opened 3:57 seconds, the progress bar on the avatar in the upper left corner shows The time it takes to open the door. 4 : 47 seconds shows that instead of staying where he is to release the resurrection spell, Whitemane moves to the nearest position 0.5 yards from the corpse to release the resurrection spell

## Tests Performed:

- windows10
- Builds without errors.


## How to Test the Changes:


1. .instance unbind all
2. .tele smcath
3. Enter the battle normally Pull Whitemane away from Mograine's corpse After hypnotizing the player with Whitemane's HP below 50%, observe if Whitemane will wait in a daze
4.~~Will the resurrection of Mograine use the "salute" emoji only after resurrection~~
5. Whether Mograine will use SPELL_LAY_ON_HANDS (note the spell release action) to treat Whitemane after being revived, ~~and whether the "roar" emoji will be used after use~~
6.  Kill Whitemane and Mograine survive to break away from the fight. Observe if the mograine will be reset again
7. Whitemane should not die if he takes excessive damage before releasing the resurrection spell (HP is always 1)
8. Check to see if the church door is locked, you need a key to open it
## Known Issues and TODO List:


- [ ]Whitemane dies Mograine leaves the battle while alive Mograine runs back to the reset point and disappears after the player revives (Blizzard disappears after waiting a few seconds in place.) Do not go back to the reset point)


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
